dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/security/impl/ldap/LdapAuthenticator.java

```diff
@@ -37,7 +38,10 @@ public User login(String userId, String password, String extra) {
             //check if user exist
             user = usersService.getUserByUserName(userId);
             if (user == null) {
-                user = usersService.createUser(ldapService.getUserType(userId), userId, ldapEmail);
+                LdapUserNotExistActionType type = ldapService.getLdapUserNotExistAction();
+                if(type == LdapUserNotExistActionType.CREATE){
+                    user = usersService.createUser(ldapService.getUserType(userId), userId, ldapEmail);
+                }
```

Comment:
NIT, Since we only check whether ldap action is created now, maybe we can add a function named `createIfUserNotExists` in `ladService`  and we here can use

```suggestion
                if(ldapService.createIfUserNotExists()){
                    user = usersService.createUser(ldapService.getUserType(userId), userId, ldapEmail);
                }
```

Reply:
Yeah, It will be more elegant, Done.